src/snowflake/snowpark/mock/mock_plan.py

```diff
@@ -676,24 +711,34 @@ def calculate_expression(
         if exp.name == "array_agg":
             to_pass_args[-1] = exp.is_distinct
         if exp.name == "sum" and exp.is_distinct:
-            to_pass_args[0] = ColumnEmulator(data=to_pass_args[0].unique())
-            to_pass_args[0].sf_type = LongType()
+            to_pass_args[0] = ColumnEmulator(
+                data=to_pass_args[0].unique(), sf_type=to_pass_args[0].sf_type
+            )
         return _MOCK_FUNCTION_IMPLEMENTATION_MAP[exp.name](*to_pass_args)
     if isinstance(exp, ListAgg):
         column = calculate_expression(exp.col, input_data, analyzer, expr_to_alias)
+        column.sf_type = ColumnType(ArrayType(), exp.col.nullable)
```

Comment:
ListAgg returns a string type from the document: https://docs.snowflake.com/en/sql-reference/functions/listagg#returns